lib/iris/analysis/_scipy_interpolate.py

```diff
@@ -166,10 +204,76 @@ def __call__(self, xi, method=None):
                     raise ValueError("One of the requested xi is out of "
                                      "bounds in dimension %d" % i)
 
-        indices, norm_distances, out_of_bounds = self._find_indices(xi.T)
+        method = self.method if method is None else method
+        prepared = (xi_shape, method) + self._find_indices(xi.T)
+
+        if SPARSE and method == 'linear':
+
+            xi_shape, method, indices, norm_distances, out_of_bounds = prepared
+
+            # Allocate arrays for describing the sparse matrix.
+            n_src_values_per_result_value = 2 ** ndim
+            n_result_values = len(indices[0])
+            n_non_zero = n_result_values * n_src_values_per_result_value
+            weights = np.ones(n_non_zero, dtype=norm_distances[0].dtype)
+            col_indices = np.empty(n_non_zero)
+            row_ptrs = np.arange(0, n_non_zero + n_src_values_per_result_value,
+                                 n_src_values_per_result_value)
+
+            corners = itertools.product(*[[(i, 1 - n), (i + 1, n)]
```

Comment:
@rhattersley - this looks like it could be slow. Did you profile this part? Looks to me like there might be room for an optimisation in the future :wink: